src/parquet/bloom_filter_algorithm.cc

```diff
@@ -16,11 +16,40 @@
 // under the License.
 
 #include "parquet/bloom_filter_algorithm.h"
+#include "arrow/status.h"
+#include "arrow/util/bit-util.h"
+#include "parquet/exception.h"
 #include "parquet/types.h"
 
 namespace parquet {
 constexpr uint32_t BlockBasedAlgorithm::SALT[BITS_SET_PER_BLOCK];
 
+BlockBasedAlgorithm::BlockBasedAlgorithm(uint32_t num_bytes)
```

Comment:
This is getting complex, which is why I suggest, as I mentioned a few times before, having two classes:

`Filter`, which is has pure abstract virtual methods, and `BlockSplitBloomFilter`, but not `BloomFilterAlgorithm` class.

```
struct Filter {
  virtual void InsertHash(uint64_t) = 0;
  virtual void FindHash(uint64_t) const = 0;
  // Hash methods, destructor, all pure virtual
};

struct BlockSplitBloomFilter : public Filter {
  // Constructor
  // Overrides
};
```

I suggest you remove the `BloomFilterAlgorithm` class.

Reply:
Agree with you, that looks like simple. Just updated code, please help to have a look.